src/activities/Elsa.Activities.SQL/Client/SqlClient.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
```

Comment:
`Microsoft.Data.SqlClient` is specific to Microsoft SQL Server I think. We need to abstract the client so any database can be supported (Oracle, SQLite, MySQL etc.)

Reply:
There are a number of ways to look at this.

1. This activity is a low-level database activity that targets a specific engine (SQL Server). This means we could add multiple activities that target a specific database engine.
2. This activity is a high-level database activity that can target any supported engine. The user would have to choose the desired target engine from a dropdown list.
3. A variation on option 2, but instead of letting the user select from a dropdown list, there would be an activity type provider that dynamically adds database activities based on configured database engine client providers.

One nice advantage of the first option is that the activity can potentially expose database engine-specific settings. But if we think we never need specialized settings for a given database engine (and they all just need a connection string), then option 2 might be the sweet spot.

@andrewkorol @jwillmer Any thoughts on this?

Reply:
I think for now this will cover option one (since we require it) but I agree that it would be a nice enhancement to support more. I guess the best option is to wait for user demand and granular improve it.

Reply:
> There are a number of ways to look at this.
> 
> 1. This activity is a low-level database activity that targets a specific engine (SQL Server). This means we could add multiple activities that target a specific database engine.
> 2. This activity is a high-level database activity that can target any supported engine. The user would have to choose the desired target engine from a dropdown list.
> 3. A variation on option 2, but instead of letting the user select from a dropdown list, there would be an activity type provider that dynamically adds database activities based on configured database engine client providers.
> 
> One nice advantage of the first option is that the activity can potentially expose database engine-specific settings. But if we think we never need specialized settings for a given database engine (and they all just need a connection string), then option 2 might be the sweet spot.
> 
> @andrewkorol @jwillmer Any thoughts on this?

I agree that as a start we can make it low level and specific to SQL server. Maybe the activity can be renamed to SqlServer to avoid confusion.

I don't think there will be any further customization needed for the different databases though. All will need to select a driver, provide a connection string and a query. Differences might occur if we introduce bluk activities in the future.

This activity will have good potential for future expansion to support SQL syntax with paramerized queries, support bulk operations, and so on. I really appreciate that it already started.

Reply:
> Maybe the activity can be renamed to SqlServer to avoid confusion.

I was thinking the same thing. Then we can always later add multiple flavors, and/or add a "generic" version from which the user can select a provider from a dropdown list. At the very least, renaming to e.g. "SqlServer" gives us these options.

Also, the activity should probably be named a little bit more specifically, because what it's doing is: running a SQL query.
So the name should probably something like: `ExecuteSqlServerQuery` (a verb). The term `SqlServer` is just a noun and doesn't convey what it's doing.